incubator/mongodb/templates/mongodb-petset.yaml

```diff
@@ -0,0 +1,136 @@
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  serviceName: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  replicas: {{default 3 .Values.Replicas | quote }}
+  template:
+    metadata:
+      labels:
+        heritage: {{.Release.Service | quote }}
+        release: {{.Release.Name | quote }}
+        chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+        component: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+      annotations:
+        "helm.sh/created": {{.Release.Time.Seconds | quote }}
+        pod.alpha.kubernetes.io/initialized: "true"
+        pod.alpha.kubernetes.io/init-containers: '[
+            {
+                "name": "install",
+                "image": "gcr.io/google_containers/mongodb-install:0.1",
+                "args": ["--work-dir=/work-dir"],
+                "volumeMounts": [
+                    {
+                        "name": "workdir",
+                        "mountPath": "/work-dir"
+                    },
+                    {
+                        "name": "config",
+                        "mountPath": "/config"
+                    }
+                ]
+            },
+            {
+                "name": "bootstrap",
+                "image": "{{.Values.Image}}:{{.Values.ImageTag}}",
+                "command": ["/work-dir/peer-finder"],
+                "args": ["-on-start=\"/work-dir/on-start.sh\"", "-service={{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"],
+                "env": [
+                  {
+                      "name": "POD_NAMESPACE",
+                      "valueFrom": {
+                          "fieldRef": {
+                              "apiVersion": "v1",
+                              "fieldPath": "metadata.namespace"
+                          }
+                      }
+                   }
+                ],
+                "volumeMounts": [
+                    {
+                        "name": "workdir",
+                        "mountPath": "/work-dir"
+                    },
+                    {
+                        "name": "config",
+                        "mountPath": "/config"
+                    },
+                    {
+                        "name": "datadir",
+                        "mountPath": "/data/db"
+                    }
+                ]
+            }
+        ]'
+    spec:
+      containers:
+      - name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+        image: "{{.Values.Image}}:{{.Values.ImageTag}}"
+        imagePullPolicy: "{{.Values.ImagePullPolicy}}"
+        ports:
+        - containerPort: {{.Values.PeerPort}}
+          name: peer
+        resources:
+          requests:
+            cpu: "{{.Values.Cpu}}"
+            memory: "{{.Values.Memory}}"
+        command:
+        - /usr/bin/mongod
+        - --config=/config/mongod.conf
+        readinessProbe:
```

Comment:
I think I understand what is attempting to be done, but my point is that it may not be the best approach.  PetSet is designed to ensure that one pet starts after another, as starting distrbuted systems starting at the same time is usually not good.  There is an open issue for master slave election, which is I think what is needed.

How in a non-k8s deployment is master slave election handled?